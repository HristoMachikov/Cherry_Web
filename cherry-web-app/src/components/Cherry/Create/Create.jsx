import React, { Component } from 'react';
import { Link } from 'react-router-dom';
// import './shared/styles/_forms.scss';

const camelCased = myString => (
    myString.replace(/-([a-z])/g, (g) => g[1].toUpperCase())
);

class Create extends Component {
    state = {
        sort: "",
        price: "",
        imagePath: "",
        description: "",
        checkBox: "",
        isPublic: true,
        errorMessages: []
    };

    handleCreate = (event) => {
        event.preventDefault();
        console.dir(this.state);
    }

    handleFormElementChange = (event) => {
        const { value, id } = event.target;
        const parsedId = camelCased(id);
        this.setState({
            [parsedId]: value
        })
    }

    checkValidity = (event) => {
        const { target } = event;
        if (!target.checkValidity()) {
            console.error("Something went wrong...")
            console.error(target.validationMessage)
            this.setState(({ errorMessages }) => ({
                errorMessages: [target.validationMessage]
            }))
        } else {
            this.setState(() => ({
                errorMessages: []
            }))
        }
    }

    render() {
        const { sort, price, imagePath, description, checkBox, errorMessages, isPublic } = this.state;
        return (
            <section className="contacts">
                <div className="contact-header">
                    <h1>Създай нов сорт</h1>
                </div>
                {
                    errorMessages.length ? <ul>
                        {
                            errorMessages.map((message, idx) => <li key={idx}>{message}</li>)
                        }
                    </ul> : null
                }
                <div className="contact-form">
                    <form onSubmit={this.handleCreate}
                    // action='/cherry/create'
                    // method="POST"
                    >

                        <p className="sort">
                            <label htmlFor="sort">Име:<span>*</span></label>
                            <input
                                type="text"
                                name="sort"
                                id="sort"
                                placeholder="Type cherry sort here"
                                value={sort}
                                onChange={this.handleFormElementChange}
                                required
                                onBlur={this.checkValidity}
                            />
                        </p>
                        <p className="price">
                            <label htmlFor="price">Цена:<span>*</span></label>
                            <input
                                type="text"
                                name="price"
                                id="price"
                                placeholder="Type price per 1 kilogram"
                                value={price}
                                onChange={this.handleFormElementChange}
                                required
                                onBlur={this.checkValidity}
                            />
                        </p>
                        <p className="imagePath">
                            <label htmlFor="image-path">Снимка:<span>*</span></label>
                            <input
                                type="text"
                                name="imagePath"
                                id="image-path"
                                placeholder="Type image path"
                                value={imagePath}
                                onChange={this.handleFormElementChange}
                                required
                                onBlur={this.checkValidity}
                            />
                        </p>
                        <p className="description">
                            <label htmlFor="description">Съобщение:<span>*</span></label>
                            <textarea
                                name="description"
                                id="description"
                                rows="5"
                                placeholder="Type some description..."
                                value={description}
                                onChange={this.handleFormElementChange}
                                required
                                onBlur={this.checkValidity}

                            />
                        </p>
                        <p className="check-box">
                            <label htmlFor="check-box">Public</label>
                            <input
                                type="checkbox"
                                className="checkbox"
                                id="check-box"
                                name="checkBox"
                            // {isPublic ? "checked" : ""} 
                            />
                        </p>
                        <div className="form-btn">
                            <p className="btn">
                                <button type="submit" className="primary-btn">Създай</button>
                            </p>
                        </div>


                        {/* <fieldset className="main-form-fieldsed">
                            <legend className="main-form-legent">Вход</legend>
                            <p className="form-field username">
                                <label htmlFor="username">Име</label>
                                <input
                                    className="form-input"
                                    type="text"
                                    name="username"
                                    id="username"
                                    value={username}
                                    onChange={this.handleFormElementChange}
                                    required
                                    onBlur={this.checkValidity}
                                />
                                <span></span>
                            </p>
                            <p className="form-field password last">
                                <label htmlFor="password">Парола</label>
                                <input
                                    className="form-input"
                                    type="password"
                                    name="password"
                                    id="password"
                                    value={password}
                                    onChange={this.handleFormElementChange}
                                    required
                                    onBlur={this.checkValidity}
                                />
                                <span></span>
                            </p>
                            <p className="form-btn">
                                <button className="primary-btn" type="submit">Вход</button>
                            </p>
                            <p className="question">Имаш ли <Link to="/user/register">Регистрация</Link>?</p>
                        </fieldset> */}
                    </form>
                </div>
            </section >
        );
    }

}

export default Create;