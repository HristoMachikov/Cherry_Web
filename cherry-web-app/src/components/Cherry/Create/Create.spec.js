import React from 'react';
import { shallow } from 'enzyme';
import Create from './Create';
import cherryService from '../../../services/cherry-service';

describe('Create component', () => {
    it('should create correct component', () => {
        const wrapper = shallow(<Create />);
        expect(wrapper).toBeTruthy();
    })
    it('should checks for html changes', () => {
        const wrapper = shallow(<Create />);
        expect(wrapper).toMatchSnapshot();
    })
    // it("should create sort successfully", (done) => {
    //     const history = { push: () => {} };
    //     // const useRefSpy = jest.spyOn(React, 'useRef').mockReturnValueOnce({ current: { value: 123 } });
    //      const spy = jest.spyOn().mockReturnValueOnce({ 
    //          sort: { value: 123 },
    //          price: { value: 1 },
    //          imagePath: { value: 1234 },
    //          description: { value: 12 },
    //          checkBox: { value: false }
    //          });
    //     const historyPushSpy = jest.spyOn(history, 'push');
    //     const createSortSpy = jest.spyOn(cherryService, 'postCreate').mockImplementationOnce(() => Promise.resolve());;
      
    //     const wrapper = shallow(<Create history={history}/>);
    //     wrapper.find('button').simulate('click');
      
    //     expect(spy).toHaveBeenCalledTimes(1);
    //     expect(createSortSpy).toHaveBeenCalledTimes(1);
    //     // expect(createSortSpy).toHaveBeenCalledWith({ description: 123 });
    //     Promise.resolve().then(() => {
    //       expect(historyPushSpy).toHaveBeenCalledTimes(1);
    //       done();
    //     //   useRefSpy.mockClear();
    //     });
    //   });
})