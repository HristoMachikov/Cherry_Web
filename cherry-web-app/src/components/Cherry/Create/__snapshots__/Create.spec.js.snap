// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create component should checks for html changes 1`] = `
<section
  className="contacts"
>
  <ToastContainer
    autoClose={3500}
    bodyClassName={null}
    className={null}
    closeButton={
      <CloseButton
        ariaLabel="close"
      />
    }
    closeOnClick={true}
    draggable={true}
    draggablePercent={80}
    hideProgressBar={false}
    newestOnTop={false}
    pauseOnFocusLoss={true}
    pauseOnHover={true}
    position="top-right"
    progressClassName={null}
    progressStyle={null}
    role="alert"
    rtl={false}
    style={null}
    toastClassName={null}
    transition={[Function]}
  />
  <div
    className="contact-header"
  >
    <h1>
      Създай нов сорт
    </h1>
  </div>
  <div
    className="contact-form"
  >
    <form
      onSubmit={[Function]}
    >
      <p
        className="sort"
      >
        <label
          htmlFor="sort"
        >
          Име:
          <span>
            *
          </span>
        </label>
        <input
          id="sort"
          name="sort"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Type cherry sort here"
          required={true}
          type="text"
          value=""
        />
      </p>
      <p
        className="price"
      >
        <label
          htmlFor="price"
        >
          Цена:
          <span>
            *
          </span>
        </label>
        <input
          id="price"
          name="price"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Type price per 1 kilogram"
          required={true}
          type="text"
          value=""
        />
      </p>
      <p
        className="imagePath"
      >
        <label
          htmlFor="image-path"
        >
          Снимка:
          <span>
            *
          </span>
        </label>
        <input
          id="image-path"
          name="imagePath"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="/images/imgName.jpg"
          required={true}
          type="text"
          value=""
        />
      </p>
      <p
        className="description"
      >
        <label
          htmlFor="description"
        >
          Съобщение:
          <span>
            *
          </span>
        </label>
        <textarea
          id="description"
          name="description"
          onBlur={[Function]}
          onChange={[Function]}
          placeholder="Type some description up to 80 symbols"
          required={true}
          rows="5"
          value=""
        />
      </p>
      <div
        className="check-box"
      >
        <label
          htmlFor="check-box"
        >
          Да се публикува
        </label>
        <input
          id="check-box"
          name="checkBox"
          onChange={[Function]}
          type="checkbox"
          value={false}
        />
      </div>
      <div
        className="form-btn"
      >
        <p
          className="btn"
        >
          <button
            className="primary-btn"
            type="submit"
          >
            Създай
          </button>
        </p>
      </div>
    </form>
  </div>
</section>
`;
